src/main/java/com/lineate/testyourlexicon/services/GameService.java

```diff
@@ -47,13 +54,116 @@ public SupportedLanguagesDto supportedLanguages() {
   }
 
   public GameInitializedDto initGameForUser(User user) {
-    if (gameRepository.getUserActiveGame(user).isPresent()) {
-      throw new GeneralMessageException("A game is already started for the user");
-    }
+    checkIfGameActiveForUser(user);
     Game game = new Game();
     game.setUser(user);
     game.setStepsLeft(user.getGameConfiguration().getNumberOfSteps());
     gameRepository.save(game);
+    log.info("Initialized new game for user {'user': {}, 'game_id': {}}",
+        user.getId(), game.getGameId());
     return new GameInitializedDto(game.getGameId());
   }
+
+  public Question generateRandomQuestion(GameConfiguration gameConfiguration, Game game) {
+    String translateFrom = gameConfiguration.getTranslateFrom();
+    String translateTo = gameConfiguration.getTranslateTo();
+    int answerOptionsCount = gameConfiguration.getAnswerCount();
+    Question question = null;
+    do {
+      question = translationService.getRandomQuestion(translateFrom,
+        translateTo, answerOptionsCount);
+    } while (questionRepository.existsByGameAndTranslationId(game, question.getTranslationId()));
+    return question;
+  }
+
+  public QuestionEntity saveQuestionInDatabase(Question question, Game game) {
+    QuestionEntity questionEntity = new QuestionEntity();
+    questionEntity.setGuessed(false);
+    questionEntity.setTranslationId(question.getTranslationId());
+    questionEntity.setGame(game);
+    return questionRepository.save(questionEntity);
+  }
+
+  public Game updateGameCurrentQuestion(Game game, QuestionEntity questionEntity) {
+    game.setStepsLeft(game.getStepsLeft() - 1);
+    game.setCurrentQuestionId(questionEntity.getId());
+    return gameRepository.save(game);
+  }
+
+  public StepDto userActiveGameNextStep(User user, long gameId) {
+    Game game = validateGameForUser(user, gameId);
+    Question question = generateRandomQuestion(user.getGameConfiguration(), game);
+    QuestionEntity questionEntity = saveQuestionInDatabase(question, game);
+    game = updateGameCurrentQuestion(game, questionEntity);
```

Comment:
It is a good practice to avoid variables' mutations if possible. Let's declare a new variable here:

```suggestion
    final var updatedGame = updateGameCurrentQuestion(game, questionEntity);
```